### PR TITLE
use sql query to check freshness of Products

### PR DESCRIPTION
**Pull request by snopoke**
See discussion on https://github.com/dimagi/commcare-hq/pull/24918

### CODE REVIEW

File: corehq/apps/products/fixtures.py [hunk 1]
```diff
@@ -113,5 +119,20 @@ def get_products():
...
+        last_sync = restore_state.last_sync_log
+        if not last_sync:
+            # definitely sync if we haven't synced before
+            return True
+
+        changes_since_last_sync = SQLProduct.objects.filter(
+            domain=restore_state.restore_user.domain, last_modified__gte=last_sync.date
+        ).exists()
+
+        return changes_since_last_sync
```
Comment (mkangia) [inline corehq/apps/products/fixtures.py]: nit: `changed_since_last_sync`

File: corehq/apps/programs/fixtures.py [hunk 2]
```diff
@@ -43,13 +43,37 @@ def __call__(self, restore_state):
...
+        """
+        Determine if a data collection needs to be synced.
+        """
+
+        # definitely sync if we haven't synced before
+        if not last_sync or not last_sync.date:
+            return True
+
+        # check if any items have been modified since last sync
+        for data_item in data:
```
Comment (mkangia) [inline corehq/apps/programs/fixtures.py]: how many entries would this be?
If this can be high,
are you considering making a couch view for this?

**Review by mkangia — COMMENTED** [review]
couple of queries. Defer to Emord for a review from decision perspective.

File: corehq/apps/programs/fixtures.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/programs/fixtures.py]: Generally very view

**Comment by snopoke** [comment]
:white_check_mark: @emord

**Pull request merged**